src/Raven.Server/Documents/Indexes/Index.cs

```diff
@@ -823,6 +857,9 @@ protected void ExecuteIndexing()
                                 }
                                 catch (OperationCanceledException)
                                 {
+                                    // We are here only in the case of indexing process cancellation.
+                                    Debug.Assert(!_batchProcessCancellationTokenSource.IsCancellationRequested);
```

Comment:
This is run before that: line 788 re-throws a OperationCanceledException, which will be caught by this handler. After this handler runs, the finally block from line 791 will run, setting this variable to null.